src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceService.State.cs

```diff
@@ -0,0 +1,297 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.SyncNamespace
+{
+    internal abstract partial class AbstractSyncNamespaceService<TNamespaceDeclarationSyntax, TCompilationUnitSyntax>
+        where TNamespaceDeclarationSyntax : SyntaxNode
+        where TCompilationUnitSyntax : SyntaxNode
+    {
+        internal sealed class State
+        {                                                                        
+            private static readonly SymbolDisplayFormat s_qualifiedNameOnlyFormat = 
+                new SymbolDisplayFormat(globalNamespaceStyle: SymbolDisplayGlobalNamespaceStyle.Omitted,
+                                        typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces);
+
+            public Solution Solution { get; }
+
+            public DocumentId OriginalDocumentId { get; }
+
+            public ImmutableArray<DocumentId> DocumentIds { get;  }
+
+            /// <summary>
+            /// This is the default namespace defined in the project file.
+            /// </summary>
+            public string DefaultNamespace { get; }
+
+            /// <summary>
+            /// This is the name of the namespace declaration that trigger the refactoring.
+            /// </summary>
+            public string DeclaredNamespace { get; }
+                                                                                                       
+            /// <summary>
+            /// This is the new name we want to change the namespace to.
+            /// Empty string means global namespace, whereas null means renaming action is not available.
+            /// </summary>
+            public string TargetNamespace { get; }
+
+            /// <summary>
+            /// This is the part of the declared namespace that is contained in default namespace.
+            /// We will use this to construct target folder to move the file to.
+            /// For example, if default namespace is `A` and declared namespace is `A.B.C`, 
+            /// this would be `B.C`.
+            /// </summary>
+            public string RelativeDeclaredNamespace { get; }
```

Comment:
does this really need to be a property?  it seems lke it is redundant with the other properties alreayd here.

Reply:
this comment still applies.

Reply:
Since it's already calculated, I don't see any problem saving and reusing it (even though the calculation is cheap), especially given that this is a private type.